develop/tutorials/articles/03-portlets/packaging-a-jsf-application.markdown

```diff
@@ -50,6 +46,51 @@ example anatomy structure is outlined below:
     - `faces-config.xml`
     - `web.xml`
 
+## Auto-Generated WAB [](id=auto-generated-wab)
+
+The source of an auto-generated JSF Portlet WAB follows the standard
+directory layout for WAR projects. Bnd directives for the manifest can be placed in the liferay-plugin-package.properties file.  For use of Declarative Services in your JSF portlet, a java class that extends [GenericFacesPortlet](http://myfaces.apache.org/portlet-bridge/2.0/api/apidocs/javax/portlet/faces/GenericFacesPortlet.html) may be included and annotated with `@Component` on the class and `@Reference` on a method to inject a service.
+
+- `src/`
+    - `main/`
+        - `java/`
+        - `webapp/`
+            - `WEB-INF/`
+                - `faces-config.xml`
+                - ~~`liferay-display.xml`~~ (no longer needed)
+                - `liferay-plugin-package.properties`
+                - ~~`liferay-portlet.xml`~~ (no longer needed)
+                - ~~`portlet.xml`~~ (no longer needed)
+                - `web.xml`
+
+**Benefits:**
+
+- Processed by the Liferay [WAB Generator](https://github.com/liferay/liferay-portal/tree/master/modules/apps/foundation/portal-osgi-web/portal-osgi-web-wab-generator), which automatically creates an OSGi-ready `META-INF/MANIFEST.MF`.
+- Processed by the Liferay auto-deploy process, which adds the `PluginContextListener` to the `WEB-INF/web.xml` descriptor when the WAB has not been pre-configured.
+- Can affect the content of `META-INF/MANIFEST.MF` by putting BND directives and OSGi headers into the `WEB-INF/liferay-plugin-package.properties` file.
```

Comment:
@vsingleton I know that OSGi headers like Bundle-Version can be put in there, but I am not sure if BND directives like "-wab" (or whatever) have meaning in there. If there is a convenient BND directive to try, please check.